src/main/java/org/design_manager_project/services/BaseService.java

```diff
@@ -38,14 +38,18 @@ public Page<DTO> findAllWithPage(FT ft){
         return baseMapper.convertPageToDTO(baseRepository.findAllWithFilter(ft.getPageable(), ft));
     }
 
-    public Optional<DTO> findById(ID id) throws NoSuchFieldException, IllegalAccessException {
+    public Optional<DTO> findById(ID id) {
         E e = baseRepository.findById(id).orElseThrow(() -> new EntityNotFoundException("Not found entity with id: " + id));
 
-        Field deletedAtField = e.getClass().getDeclaredField("deletedAt");
-        deletedAtField.setAccessible(true);
-        Object deletedAtValue = deletedAtField.get(e);
-        if (deletedAtValue != null) {
-            throw new BadRequestException(Constants.OBJECT_DELETED);
+        try {
+            Field deletedAtField = e.getClass().getDeclaredField("deletedAt");
```

Comment:
It is possible to create a separate reusable function to check if a record has been deleted. For example, when updating or deleting a record that has already been deleted previously.